lib/src/query_editor_view.dart

```diff
@@ -0,0 +1,564 @@
+import 'package:ditto_live/ditto_live.dart';
+import 'package:flutter/material.dart';
+
+import 'query_result_exporter.dart';
+
+class QueryEditorView extends StatefulWidget {
+  final Ditto ditto;
+
+  const QueryEditorView({
+    super.key,
+    required this.ditto,
+  });
+
+  @override
+  State<QueryEditorView> createState() => _QueryEditorViewState();
+}
+
+class _QueryEditorViewState extends State<QueryEditorView> {
+  final TextEditingController _queryController = TextEditingController();
+  final TextEditingController _pageController = TextEditingController();
+  bool _isExecuting = false;
+  List<String> _allResults = [];
+  List<String> _paginatedResults = [];
+  String? _errorMessage;
+  dynamic _rawQueryResult;
```

Comment:
We should really never be using `dynamic` as an explicit type without an incredibly strong justification. `Map<String, dynamic>` specifically for interacting with JSON code is an exception

Reply:
Looks like `QueryResult?` here is fine?